crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -227,6 +227,22 @@ impl UpdateTask {
 }
 
 impl CachingNativePriceEstimator {
+    /// Convert from normalized price to floating point price
+    fn from_normalized_price(price: U256) -> Option<f64> {
+        // Convert U256 to f64
+        let price_in_eth = price.to_f64_lossy();
+
+        // Divide by 1e18 to reverse the multiplication by 1e18
+        let normalized_price = price_in_eth / 1e18;
+
+        // Ensure the price is in the normal float range
+        if normalized_price.is_normal() && normalized_price >= 1e-18 {
```

Comment:
I don't think there is a valid range of prices here. The `f64` native price of a token depends on the value of the token relative to the reference price (ETH/DAI in our cases) and on the number of decimals of these tokens. So I think this can actually just be:
```rust
normalized_price.is_normal().then_some(normalized_price)
```